src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -516,17 +517,21 @@ public DatasetSummary mapRow(ResultSet rs, int rowNum) throws SQLException {
 
     /**
      * Probe to see if can access database
+     *  @return status and if failure, exception message in RepositoryStatusModelSystems model
      */
-    public boolean statusCheck() {
+    public RepositoryStatusModelSystems statusCheck() {
         String sql = "SELECT count(1)";
         MapSqlParameterSource params = new MapSqlParameterSource();
+        jdbcTemplate.queryForObject(sql, params, Integer.class);
         try {
             jdbcTemplate.queryForObject(sql, params, Integer.class);
-            return true;
+            return new RepositoryStatusModelSystems()
+                .ok(true);
         } catch (Exception ex) {
-            logger.error("Database status check failed: " + ex.getMessage());
-            return false;
+            logger.error("Database status check failed with Exception: {}", ex);
```

Comment:
I don't think this will work the way you expect, I think this will call `error(String, Throwable)` not `error(String, Object)`. Which is OK, but the `{}` won't be used.
```suggestion
            logger.error("Database status check failed", ex);
```